plugins/woocommerce/templates/cart/cart.php

```diff
@@ -85,23 +87,9 @@
 						<td class="product-name" data-title="<?php esc_attr_e( 'Product', 'woocommerce' ); ?>">
 						<?php
 						if ( ! $product_permalink ) {
-							/**
-							 * Filter the product name.
-							 *
-							 * @since 7.8.0
-							 * @param string $product_name Name of the product in the cart.
-							 * @param array $cart_item The product in the cart.
-							 * @param string $cart_item_key Key for the product in the cart.
-							 */
-							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', $product_name, $cart_item, $cart_item_key ) . '&nbsp;' );
+							echo wp_kses_post( $product_name . '&nbsp;' );
```

Comment:
Solves the problem ... a little nervous because it means that callbacks on `woocommerce_cart_item_name` may be fired twice for the same item (once via line 51 and again via line 92, unless the permalink is empty), but I don't know we can do too much about that now the product name is used in the remove item link.